### PR TITLE
Move variable font output to separate folder, add test cases to fontmake

### PR DESCRIPTION
**Pull request by zjusbo**
#199

### CODE REVIEW

**Comment by zjusbo** [comment]
Some questions:

1. How should we test the python version in our code base. I used try/except. The other option is check sys.version_info_

2. What does 'interpolatable' mean, can this parameter coexist with 'variable'?

   def _output_dir(self, ext, is_instance=False, interpolatable=False,
                    autohinted=False, is_variable=False):

File: Lib/fontmake/font_project.py [hunk 1]
```diff
@@ -453,30 +455,52 @@ def run_from_ufos(
...
+            Returns ufo directly if ufo is a string. This case happens when
+            generating variable font, where ufo stores the output font name.
+        """
+        l = [str]
+        try:
+            # Python 3+ does not have unicode keyword.
+            l.append(unicode)
+        except NameError:
+            pass
+        if type(ufo) in l :
```
Comment (anthrotype) [inline Lib/fontmake/font_project.py]: here you can do this: 

```python
from fontTools.misc.py23 import basestring

if isinstance(ufo, basestring):
```

**Comment by anthrotype** [comment]
> How should we test the python version in our code base

in general, checking `sys.version_info` is the right way. You can also import `PY2` and `PY3` constants from `fontTools.misc.py23` module, which come from sys.version_info anyway.

> interpolatable

This means that only master TTFs are built, but not the final variable font. The `ttf-interpolatable` format option was introduced before the `variable` one, and at that time the only way to make a variable font was to first make the master interpolatable TTFs, and then run fontTools.varLib.main script on them.

---

Bo, I actually don't think we need an extra folder for storing the output variable font.
It's never going to be more than one file anyway, what's the point in creating a folder then?

If you also find confusing that the generated *-VF.ttf ends up next in `master_ufo` folder, then we can just change that and save the variable font in the current folder.

BTW, I've always found fontmake's use of hard-coded destination folders ("master_ufo", "instance_ttf") not ideal. I wish these could be customized somehow. The `-o` option for specifying the output _format_ instead of the _destination_ is quite unusual, I believe. But these are probably different issues.

**Comment by verbosus** [comment]
I have a patch for specifying the output folder in general (if I rememer right I added `-d --directory` since `-o --output` was already used), give me some time to clean it up.

File: setup.py [hunk 2]
```diff
@@ -43,6 +43,7 @@
         "MutatorMath>=2.0.2",
         "defcon>=0.2.1",
         "booleanOperations>=0.6.4",
+        "mock>=2.0.0",
```
Comment (anthrotype) [inline setup.py]: I like that you used the `mock` module to write the unit tests.

However, if anything,`mock` is only a _test_ requirement, not an _install_ requirement, so it should not be in `install_requires`.

Similarly, the `requirements.txt` file should not contain `mock` either, as that's used for specifying install requirements with pinned-down version; if we want the tests to be run with fixed versions of the test requirements, we could create a separate `test-requirements.txt` file or similar.

File: test/test_arguments.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+from mock import patch
+import mock
+from fontTools.ttLib import TTFont
+from fontmake.font_project import FontProject
+import fontmake.__main__ as entry
+import fontTools
+
+
+class TestFunctionsAreCalledByArguments(unittest.TestCase):
+  @patch('fontmake.font_project.FontProject.run_from_glyphs')
```
Comment (anthrotype) [inline test/test_arguments.py]: we use four spaces indentation everywhere else; this file uses two.

File: test/test_output.py [hunk 4]
```diff
@@ -18,7 +18,6 @@
 
 from fontTools.ttLib import TTFont
 
-
```
Comment (anthrotype) [inline test/test_output.py]: removing this line is not necessary. Keeping two empty lines between module-scope classes or function definitions is recommended in PEP8.

File: Lib/fontmake/font_project.py (hunk 1 above)
Comment (zjusbo) [inline Lib/fontmake/font_project.py]: Thanks Cosimo. I'll utilize misc.py23 more.

File: setup.py (hunk 2 above)
Comment (zjusbo) [inline setup.py]: I created a test-requriements.txt file and probably I'm changing our test engine config.

File: test/test_arguments.py (hunk 3 above)
Comment (zjusbo) [inline test/test_arguments.py]: Thans Cosimo. Done.

File: test/test_output.py (hunk 4 above)
Comment (zjusbo) [inline test/test_output.py]: Oh Thanks for pointing the PEP8 style guide. It's worth reading and I looked though it. I made some style changes in the codebase based on its recommendation.

**Comment by zjusbo** [comment]
@anthrotype Is it posible for user to create multiple VF fonts by running a script? I was thinking in @marekjez86 's use case. He may create many VF fonts at one time. So probably dumping them into a single folder seems reasonable? Please correct me if I'm wrong.

@verbosus Thanks for doing it! adding '-d' in command line argument would be more user friendly.

**Pull request merged**